github/misc.go

```diff
@@ -195,3 +196,33 @@ func (c *Client) ListServiceHooks() ([]*ServiceHook, *Response, error) {
 
 	return *hooks, resp, err
 }
+
+// ParseURL parses a GitHub URL and returns the owner and repo (if provided).
+// It returns an error if it is unable to parse the URL.
+//
+// Examples of valid URLs are:
+//     https://github.com/bazelbuild/bazel/tree/master
+//     http://github.com/bazelbuild/bazel/tree/master
+//     https://www.github.com/bazelbuild/bazel/tree/master
+//     https://github.com/bazelbuild/bazel/releases/tag/0.2.1
+//     https://github.com/bazelbuild/bazel/commit/19b5675725caf69008a717082149237400260edc
+//     https://github.com/bazelbuild/bazel/archive/0.2.1.zip
```

Comment:
It turns out that the examples you listed were already in the tests... but I'll move them to the top so that they are easier to see (and I'll embed the wacky `////` version in the middle).